minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java

```diff
@@ -722,14 +717,15 @@ public MultiProcessCluster build() {
           "The worker indexes in worker properties should be bigger or equal to zero "
               + "and small than %s", mNumWorkers);
       return new MultiProcessCluster(mProperties, mMasterProperties, mWorkerProperties,
-          mNumMasters, mNumWorkers, mClusterName, mDeployMode);
+          mNumMasters, mNumWorkers, mClusterName, mDeployMode, mReservedPorts);
     }
   }
 
   /**
+   * @param reservedPorts ports reserved for usage by this cluster
    * @return a new builder for an {@link MultiProcessCluster}
    */
-  public static Builder newBuilder() {
-    return new Builder();
+  public static Builder newBuilder(List<ReservedPort> reservedPorts) {
+    return new Builder(reservedPorts);
```

Comment:
Should we add a `setReservedPorts` method instead of adding this here?

Reply:
I want it to be required by the builder API

Reply:
Makes sense, thanks.